spec-tests/src/test/java/org/candlepin/spec/imports/ImportCleanupSpecTest.java

```diff
@@ -0,0 +1,139 @@
+                ProductAttributes.VirtualLimit.withValue("unlimited")))
+            .derivedProduct(productDc);
+        ExportGenerator exportGenerator = generateWith(generator -> generator
+            .withProduct(productDc)
+            .withProduct(productVdc));
+
+        Export export = exportGenerator.export();
+        ConsumerDTO consumer = admin.consumers().createConsumer(Consumers.random(owner));
+        List<EntitlementDTO> ents = admin.entitlements()
+            .listAllForConsumer(export.consumer().getUuid(), null, null, null, null, null, null);
+        for (EntitlementDTO ent : ents) {
+            admin.entitlements().unbind(ent.getId());
+        }
+        // We need to wait here for unbind to finish
+        Thread.sleep(500L);
+        Export updatedExport = exportGenerator.export();
+
+        doImport(owner.getKey(), export.file());
+        // NORMAL pool
+        List<PoolDTO> normalPools = admin.pools()
+            .listPools(owner.getId(), null, productVdc.getId(), null, null, null, null, null, null);
+        assertThat(normalPools)
+            .hasSize(1);
+
+        // UNMAPPED GUEST POOL
+        List<PoolDTO> unmappedPools = admin.pools()
+            .listPools(owner.getId(), null, productDc.getId(), true, null, null, null, null, null)
+            .stream()
+            .filter(pool -> pool.getType().equals("UNMAPPED_GUEST")).collect(Collectors.toList());
+        assertThat(unmappedPools)
+            .hasSize(1);
+
+        // Consume NORMAL Pool
+        JsonNode jsonNode = admin.consumers().bindPool(consumer.getUuid(), normalPools.get(0).getId(), 1);
+        assertThat(jsonNode)
+            .hasSize(1);
+
+        // STACK DERIVED POOL is created
+        List<PoolDTO> stack = admin.pools()
+            .listPools(owner.getId(), null, productDc.getId(), null, null, null, null, null, null)
+            .stream()
+            .filter(pool -> pool.getType().equals("STACK_DERIVED")).collect(Collectors.toList());
+        assertThat(stack)
+            .hasSize(1);
+
+        doImport(owner.getKey(), updatedExport.file());
+        normalPools = admin.pools().listPools(owner.getId(), null, null, null, null, null, null, null, null);
+        assertThat(normalPools)
+            .hasSize(0);
+    }
+
+    public ExportGenerator generateWith(Consumer<ExportGenerator> setup) {
+        ExportGenerator exportGenerator = new ExportGenerator(admin);
+        setup.accept(exportGenerator.minimal());
+        return exportGenerator;
+    }
+
+    private void doImport(String ownerKey, File export) {
+        importAsync(ownerKey, export, List.of());
+    }
+
+    private void importAsync(String ownerKey, File export, List<String> force) throws ApiException {
+        AsyncJobStatusDTO importJob = admin.owners().importManifestAsync(ownerKey, force, export);
+        importJob = admin.jobs().waitForJob(importJob);
+        assertThatJob(importJob).isFinished();
```

Comment:
Same issue here as the other test: Can we make sure that the job isn't just finished, but it succeeded?

Reply:
done, but only with the first import, because second one has status SUCCESS WITH WARNING, but in the ruby test we wasn't testing that too with the second import.